File: Java/Program/Customer.java
public class Customer extends Person { // Subclass from Person
    // Attributes
    private String customerId;
    private String handphoneNumber;
    private String favoriteGroup;
    // Constructors
    public Customer() {}
    // Parameterized Constructor
    public Customer(String name, int age, String gender, String customerId,
                    String handphoneNumber, String favoriteGroup) {
        super(name, age, gender);
        this.customerId = customerId;
        this.handphoneNumber = handphoneNumber;
        this.favoriteGroup = favoriteGroup;
    }
    // Getters & Setters
    public String getCustomerId() { return customerId; }
    public void setCustomerId(String customerId) { this.customerId = customerId; }

    public String getHandphoneNumber() { return handphoneNumber; }
    public void setHandphoneNumber(String handphoneNumber) { this.handphoneNumber = handphoneNumber; }

    public String getFavoriteGroup() { return favoriteGroup; }
    public void setFavoriteGroup(String favoriteGroup) { this.favoriteGroup = favoriteGroup; }
    // Method to print customer details
    public void printCustomer() {
        System.out.printf("%s\t%s\t%s\t%d\t%s%n",
            getCustomerId(), getHandphoneNumber(), getName(), getAge(), getFavoriteGroup());
    }
}

File: CPP/Program/Customer.cpp
#pragma once
#include <iostream>
#include <string>
#include "Person.cpp"

// This class represents a Customer, inheriting from Person
class Customer : public Person {
    // Attributes specific to Customer
    private:
        string customerId;
        string handphoneNumber;
        string favoriteGroup;

    public:
        // Default constructor
        Customer(){
        }

        // Parameterized constructor
        Customer(string name, int age, string gender, string customerId, string handphoneNumber, string favoriteGroup)
            : Person(name, age, gender) {
            this->customerId = customerId;
            this->handphoneNumber = handphoneNumber;
            this->favoriteGroup = favoriteGroup;
        }

        // Setters and Getters
        void setcustomerId(string customerId){
            this->customerId = customerId;
        } 

        string getcustomerId(){
            return this->customerId;
        }

        void sethandphoneNumber(string handphoneNumber){
            this->handphoneNumber = handphoneNumber;
        } 

        string gethandphoneNumber(){
            return this->handphoneNumber;
        }

        void setfavoriteGroup(string favoriteGroup){
            this->favoriteGroup = favoriteGroup;
        } 

        string getfavoriteGroup(){
            return this->favoriteGroup;
        }

        ~Customer(){
        }
        
        // For printing customer details
        void showDetails() override {
            cout << getcustomerId() << "\t" << gethandphoneNumber() << "\t" << getName() << "\t" << getAge() << "\t" << getfavoriteGroup() << endl;
        }
};
